### PR TITLE
Fix rpm CVE-2021-20266

### PR DESCRIPTION
**Pull request by MateuszMalisz**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Fix RPM CVE-2021-20266

###### Change Log  

- Fixed CVE-2021-20266

###### Does this affect the toolchain?  

**YES**

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2021-20266

###### Test Methodology

- Pipeline build id: xxxx

### CODE REVIEW

**Review by nicogbg — APPROVED** [review]

File: SPECS/rpm/CVE-2021-20266.patch [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+diff --git a/lib/header.c b/lib/header.c
+index 0c450bea2b..cbf944515b 100644
+--- a/lib/header.c
++++ b/lib/header.c
+@@ -1968,12 +1968,42 @@ rpmRC hdrblobInit(const void *uh, size_t uc,
+ 		rpmTagVal regionTag, int exact_size,
+ 		struct hdrblob_s *blob, char **emsg)
+ {
++    size_t nb;
```
Comment (PawelWMS) [inline SPECS/rpm/CVE-2021-20266.patch:21]: I'm not sure if this is a full patch. When I read through the [bug discussion](https://bugzilla.redhat.com/show_bug.cgi?id=1927741), this is the [first version of the patch](https://github.com/rpm-software-management/rpm/pull/1500/files) and later on Todd Cullum seems to be sending an [updated patch](https://github.com/rpm-software-management/rpm/commit/8f4b3c3cab8922a2022b9e47c71f1ecf906077ef). Can you double-check?

**Review by PawelWMS — CHANGES_REQUESTED** [review]

File: SPECS/rpm/CVE-2021-20266.patch (hunk 1 above)
Comment (MateuszMalisz) [reply SPECS/rpm/CVE-2021-20266.patch:73]: You're right, I've double-checked and missed that it modifies some functions later on. I've merged the two patches.

Comment (PawelWMS) [reply SPECS/rpm/CVE-2021-20266.patch:73]: That's not exactly what I had in mind. Notice that the first version I've mentioned above has been rejected (the author has closed the PR and it was never merged). Considering that, I **believe** the updated patch is the correct one, but that's what I'd like you to double-check by closely reading the bug discussion.

Comment (christopherco) [reply SPECS/rpm/CVE-2021-20266.patch:73]: +1 here. From reading through the bug discussion, Todd Cullum's link to the [upstream patch (commit id 8f4b3c)](https://github.com/rpm-software-management/rpm/commit/8f4b3c3cab8922a2022b9e47c71f1ecf906077ef) is the only one we should use.

**Review by christopherco — APPROVED** [review]

File: SPECS/rpm/CVE-2021-20266.patch (hunk 1 above)
Comment (MateuszMalisz) [reply SPECS/rpm/CVE-2021-20266.patch]: Thanks. I looked at the patch and it did not work with our release - few of the lines seems to have changed in the meantime. I inspected the modifications and modified the patch accordingly. The modified lines were either already updated or modified in a fashion that did not affect the patch.

**Pull request merged**